### PR TITLE
Switching to the new hybrid auth flow instead of user agent flow

### PR DESCRIPTION
**Pull request by bhariharan**
In a recent core release, a new auth flow was added to support authentication for hybrid mobile apps. This new flow is backwards compatible with the existing user agent flow, but adds support for 3 new scopes on connected apps:
- `lightning`
- `content`
- `visualforce`

If connected apps include these new scopes and the accompanying mobile app requests those scopes, we will get back 7 new parameters as part of the token response:
- `lightning_domain`
- `lightning_sid`
- `visualforce_domain`
- `visualforce_sid`
- `content_domain`
- `content_sid`
- `csrf_token`

This PR replaces the existing user agent flow with the new hybrid auth flow, and stores the new parameters if they are a part of the response.

What has been tested and verified so far:
- Fresh login flow (with and without new scopes).
- Storage and retrieval of new fields, if those scopes have been requested.
- Encryption and decryption of new fields.
- Refresh token flow (with and without new scopes).

Upgrade testing:
- Tested upgrading from old flow to new flow, when the scopes were requested before and after. Upon upgrade, the first time refresh occurs, the new fields will be populated.
- Tested upgrading from old flow to new flow, when the scopes weren't requested before but were requested after. Upon upgrade, the first time refresh occurs, the new fields will still not be returned, because they weren't originally requested. The app will continue to run, and the refresh will still be successful based on the original scopes requested.

Core documentation links on the new hybrid auth flow:
- [Initial Login Flow](https://help.salesforce.com/articleView?id=sf.remoteaccess_oauth_hybrid_app_token_flow.htm&type=5)
- [Token Refresh Flow](https://help.salesforce.com/articleView?id=sf.remoteaccess_oauth_hybrid_app_refresh_token_flow.htm&type=5)

Corresponding Android PR is [here](https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2192).

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m [hunk 1]
```diff
@@ -660,7 +655,7 @@ - (NSString *)generateApprovalUrlString:(SFOAuthCredentials *)credentials {
                                           kSFOAuthDisplay, kSFOAuthDisplayTouch,
                                           kSFOAuthDeviceId,[[[UIDevice currentDevice] identifierForVendor] UUIDString]];
     
-    [approvalUrlString appendFormat:@"&%@=%@", kSFOAuthResponseType, kSFOAuthResponseTypeToken];
+    [approvalUrlString appendFormat:@"&%@=%@", kSFOAuthResponseType, kSFOAuthResponseTypeHybridToken];
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m:658]: New token flow type.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCredentials+Internal.h [hunk 2]
```diff
@@ -61,6 +61,13 @@ extern NSException * _Nullable SFOAuthInvalidIdentifierException(void);
 @property (nonatomic, readwrite, nullable) NSString *challengeString;
 @property (nonatomic, readwrite, nullable) NSString *authCode;
 @property (nonatomic, readwrite, nullable) NSMutableDictionary * credentialsChangeSet;
+@property (nonatomic, readwrite, nullable) NSString *lightningDomain;
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCredentials+Internal.h:64]: New fields.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCredentials.m [hunk 3]
```diff
@@ -327,36 +344,48 @@ - (NSURL *)overrideDomainIfNeeded{
...
-    
     if (params[kSFOAuthCommunityUrl]) {
         [self setPropertyForKey:@"communityUrl" withValue:[NSURL URLWithString:params[kSFOAuthCommunityUrl]]];
     }
-    
     if (params[kSFOAuthRefreshToken]) {
         [self setPropertyForKey:@"refreshToken" withValue:params[kSFOAuthRefreshToken]];
     }
-    
+    if (params[kSFOAuthLightningDomain]) {
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCredentials.m:368]: Setting/updating new fields during auth and refresh flows.

**Review by bbirman — APPROVED** [review]

**Review by brandonpage — APPROVED** [review]
This is more fixing code formatting that adding  new auth flow lol 👍

**Pull request merged**